utils/notebook_init_utils.py

```diff
@@ -0,0 +1,69 @@
+import pathlib
+from typing import Tuple
+
+
+def init_notebook() -> Tuple[pathlib.Path, bool]:
+    """
+    Description
+    -----------
+    Initializes the notebook environment by determining the root directory of the Git repository
+    and checking if the code is running in a Jupyter notebook.
+
+    Returns
+    -------
+    Tuple[pathlib.Path, bool]
+        - pathlib.Path: The root directory of the Git repository.
+        - bool: True if running in a Jupyter notebook, False otherwise.
+    """
+    try:
+        cfg = get_ipython().config
+        in_notebook = True
+    except NameError:
+        in_notebook = False
+
+    # Get the current working directory
+    cwd = pathlib.Path.cwd()
```

Comment:
Could the CWD change based on who calls this function? Would it make sense to expose this as a parameter instead as a result?